src/axom/lumberjack/tests/CMakeLists.txt

```diff
@@ -7,65 +7,62 @@
 #------------------------------------------------------------------------------
 
 #------------------------------------------------------------------------------
-# Specify list of tests
+# Add serial GTests based tests
 #------------------------------------------------------------------------------
+set(lumberjack_serial_tests
+    lumberjack_Lumberjack.hpp
+    lumberjack_Message.hpp
+    lumberjack_TextEqualityCombiner.hpp )
 
-# Add Serial GTests based tests
-set(gtest_lumberjack_tests
-    lumberjack_Lumberjack.cpp
-    lumberjack_Message.cpp
-    lumberjack_TextEqualityCombiner.cpp
-    )
+blt_add_executable(NAME       lumberjack_serial_tests
+                   SOURCES    lumberjack_serial_main.cpp ${lumberjack_serial_tests}
+                   OUTPUT_DIR ${TEST_OUTPUT_DIRECTORY}
+                   DEPENDS_ON lumberjack gtest 
+                   FOLDER     axom/lumberjack/tests )
 
-foreach(test ${gtest_lumberjack_tests})
-    get_filename_component( test_name ${test} NAME_WE )
-    blt_add_executable( NAME ${test_name}_test
-                        SOURCES ${test}
-                        OUTPUT_DIR ${TEST_OUTPUT_DIRECTORY}
-                        DEPENDS_ON lumberjack gtest 
-                        FOLDER axom/lumberjack/tests )
-    blt_add_test( NAME ${test_name} 
-                  COMMAND ${test_name}_test )
+foreach(test_suite ${lumberjack_serial_tests})
```

Comment:
What does the `foreach` add in this case?  Is this for a case where there are more than one serial tests?

Reply:
Each file is a test suite which contains multiple individual gtest tests.  This adds each test suite as an individual ctest test command.